webapp/src/sidepanel.tsx

```diff
@@ -70,8 +78,38 @@ export class Sidepanel extends data.Component<SidepanelProps, SidepanelState> {
         }
     }
 
+    componentDidMount(): void {
+        this.updateShouldResize();
+    }
+
     componentDidUpdate(props: SidepanelProps, state: SidepanelState) {
-        if ((this.state.height || state.height) && this.state.height != state.height) this.props.setEditorOffset();
+        if ((this.state.height || state.height) && this.state.height != state.height) {
+            this.props.setEditorOffset();
+        }
+
+        this.updateShouldResize();
+    }
+
+    componentWillUnmount(): void {
+        if (this.simResizeObserver && this.simRef) {
+            this.simResizeObserver.unobserve(this.simRef);
+        }
+
+        if (this.simPanelResizeObserver && this.simPanelRef) {
+            this.simPanelResizeObserver.unobserve(this.simPanelRef);
+        }
+    }
+
+    private updateShouldResize() {
+        const shouldResize = pxt.BrowserUtils.isTabletSize() || this.props.tutorialSimSidebar;
+        if (shouldResize != this.state.shouldResize) {
+            let height = this.state.height;
+            if (shouldResize && this.state.height === undefined && this.state.lastResizeHeight !== undefined) {
```

Comment:
Is  there a reason we want to do the explicit checks for undefined here?

Reply:
I thought height being set to 0 was kind of a weird edge case and figured it was safest to just leave it be, if that ever happened.

Reply:
Makes sense!

Reply:
Would we need to account for null here as well, though?

Reply:
aargh okay I may just do a nullish check instead

Reply:
thanks!

Reply:
No problem! I just wanted to make sure we wouldn't hit some edge case. 

For what it's worth as well, I think this would have been a case where `==` and `!=` would be safe to use.. The different types of equality is something that is still kinda hand wavey to me, though, so I find the truthy check to be a bit easier to read. That said, if you wanted to not cover the 0 case, we could go that route. I wanted to note it in general because I feel like I'm always asking about `===` and I find it ironic that the `==` is needed here.. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#loose_equality_using (the second point in the list)